src/biokbase/narrative/jobs/job.py

```diff
@@ -341,16 +330,18 @@ def _update_state(self, state: dict) -> None:
         # Check if there would be no change in updating
         # i.e., if state <= self._acc_state
         if self._acc_state is not None:
-            if {**self._acc_state, **state} == self._acc_state:
+            if merge(self._acc_state, state) == self._acc_state:
                 return
 
         state = copy.deepcopy(state)
         if self._acc_state is None:
             self._acc_state = state
         else:
-            self._acc_state.update(state)
+            merge_inplace(self._acc_state, state)
+
+        self.last_updated = time.time_ns()
```

Comment:
can you alter this so that all jobs updated that are updated from a batch of ee2 data get the same timestamp?